### PR TITLE
feat: Allow unpublishing documents

### PR DESCRIPTION
**Pull request by guilherme-diniz**
Implements the action to allow unpublish a document.
This option is only available for a publish document that wasn't archived or deleted.
Should check if the permissions for allowing whom can unpublish are set correctly.

### CODE REVIEW

File: app/menus/DocumentMenu.js [hunk 1]
```diff
@@ -225,6 +230,11 @@ class DocumentMenu extends React.Component<Props> {
               New nested document
             </DropdownMenuItem>
           )}
+          {!document.isArchived && !document.isDeleted && !document.isDraft && (
```
Comment (tommoor) [inline app/menus/DocumentMenu.js:233]:
```suggestion
          {can.unpublish && (
```

`can.unpublish` should be available here as you defined it as a policy.

File: server/models/Document.js [hunk 2]
```diff
@@ -556,6 +555,18 @@ Document.prototype.publish = async function (options) {
   return this;
 };
 
+Document.prototype.unpublish = async function (options) {
+  if (!this.publishedAt) return this.save(options);
```
Comment (tommoor) [inline server/models/Document.js:559]:
```suggestion
  if (!this.publishedAt) return this;
```

Can we just avoid the save here altogether?

File: server/policies/document.js [hunk 3]
```diff
@@ -143,3 +143,19 @@ allow(
...
+    document.collection,
+    "collection is missing, did you forget to include in the query scope?"
+  );
+
+  if (!document.publishedAt || !!document.deletedAt || !!document.archivedAt)
+    return false;
+
+  if (cannot(user, "update", document.collection)) return false;
+
+  // TODO Block unpublish a document that has nested documents
```
Comment (tommoor) [inline server/policies/document.js:158]: Good callout, I missed this on the original spec 👍

**Comment by tommoor** [comment]
closes https://github.com/outline/outline/issues/1463

**Comment by tommoor** [comment]
Sorry I forgot to mention before, can we add tests for the new api endpoint?

The existing pattern should be easy to follow

**Comment by guilherme-diniz** [comment]
Tests added!

**Comment by tommoor** [comment]
Thanks! Will test and (hopefully) merge in next 24h.

**Pull request merged**